python/fledge/plugins/south/mqtt-readings/mqtt-readings.py

```diff
@@ -301,10 +309,21 @@ def stop(self):
         self.mqtt_client.disconnect()
         self.mqtt_client.loop_stop()
 
+    def convert(self, msg):
+        constructors = [json.loads, int, float, str]
+        for constructor in constructors:
+            try:
+                # Only convert if type is string
+                converted_msg = constructor(msg) if type(msg) == str else msg
+                # Create dict if converted msg isnt already a dict
+                return converted_msg if type(converted_msg) == dict else {self.json_object_key: converted_msg}
```

Comment:
We should avoid checking here with `type` def; Instead use `isinstance(..., dict)` This will ensure OrderedDict etc. are not left.

Reply:
Ah, good point. should I also change line 317 to use isinstance for consistency?

Reply:
You may.